src/ddscxx/include/org/eclipse/cyclonedds/topic/datatopic.hpp

```diff
@@ -592,9 +592,26 @@ ddsi_serdata * serdata_from_iox_buffer(
 {
   try {
     auto d = new ddscxx_serdata<T>(typecmn, kind);
-    d->iox_chunk = iox_buffer;
-    d->iox_subscriber = sub;
 
+    if (sub == nullptr) {  // serdata from the loaned sample
+#ifndef _WIN32
+#ifndef __clang__
+# pragma GCC diagnostic push
+# pragma GCC diagnostic ignored "-Wold-style-cast"
+#endif
+#endif
+      d->iox_chunk = SHIFT_BACK_TO_ICEORYX_HEADER(iox_buffer);
```

Comment:
This has just been changed, after I merged https://github.com/eclipse-cyclonedds/cyclonedds/pull/943 and https://github.com/eclipse-cyclonedds/cyclonedds-cxx/pull/159 this morning ...

Fortunately it is a trivial change. I also suspect that updating it to use the new Iceoryx interface will eliminate the need to disable the GCC warning.